File: signup-back/app/models/enrollment/aidants_connect.rb
class Enrollment::AidantsConnect < Enrollment
  protected

  def sent_validation
    errors[:siret] << "Vous devez renseigner un SIRET d’organisation valide avant de continuer" unless nom_raison_sociale
    errors[:type_projet] << "Vous devez renseigner le type de la structure avant de continuer" unless type_projet.present?

    errors[:organization_address] << "Vous devez renseigner l’adresse de la structure avant de continuer" unless additional_content&.fetch("organization_address", false)&.present?
    if additional_content&.fetch("organization_postal_code", false)&.present?
      postal_code_regex = /^\d{5}$/
      errors[:organization_postal_code] << "Vous devez renseigner un code postal valide avant de continuer" unless postal_code_regex.match?(additional_content&.fetch("organization_postal_code", ""))
    else
      errors[:organization_postal_code] << "Vous devez renseigner le code postal de la structure avant de continuer"
    end
    errors[:organization_city] << "Vous devez renseigner la ville de la structure avant de continuer" unless additional_content&.fetch("organization_city", false)&.present?

    errors[:description] << "Vous devez renseigner la description de la démarche avant de continuer" unless description.present?

    team_members_validation("responsable_metier", "responsable Aidants Connect", true)

    errors[:cgu_approved] << "Vous devez valider les modalités d’utilisation avant de continuer" unless cgu_approved?
    unless additional_content&.fetch("has_professional_contact_only", false)&.present?
      errors[:has_professional_contact_only] << "Vous devez valider que la liste des aidants à habiliter contient exclusivement des aidants professionnels avant de continuer"
    end
    unless additional_content&.fetch("has_non_elected_contact_only", false)&.present?
      errors[:has_non_elected_contact_only] << "Vous devez valider qu’aucun élu n’est impliqué dans l’habilitation Aidants Connect"
    end
  end
end

File: frontend/src/pages/AidantsConnect.js
import React from 'react';
import Form from '../components/templates/Form';
import OrganisationSection from '../components/organisms/form-sections/OrganisationSection';
import CguSection from '../components/organisms/form-sections/CguSection';
import AidantsSection from '../components/organisms/form-sections/aidants-connect-sections/AidantsSection';
import StructureSection from '../components/organisms/form-sections/aidants-connect-sections/StructureSection';
import ÉquipeSection from '../components/organisms/form-sections/ÉquipeSection';
import { DATA_PROVIDER_PARAMETERS } from '../config/data-provider-parameters';

const initialContacts = {
  demandeur: {
    header: 'Demandeur',
    description: (
      <>
        <b>Le demandeur</b>, c'est vous, dépose la demande d’habilitation.
      </>
    ),
    forceDisable: true,
  },
  responsable_metier: {
    header: 'Responsable Aidants Connect',
    description: (
      <>
        <b>Le responsable Aidants Connect</b> est en charge de la mise en place
        et du suivi d’Aidants Connect au sein du lieu d’accueil. Il tient
        également à jour la liste des aidants habilités Aidants Connect sur son
        espace administrateur (à venir). <br />
        Pour des raisons de sécurité, nous n’acceptons que les adresses emails
        nominatives et individuelles . Étant un outil homologué par l’Agence
        Nationale de la Cohésion des Territoires, cela est un prérequis pour
        garantir la sécurité de l’outil et la fiabilité des traces de connexion.
        Seule une adresse email individuelle, professionnelle et nominative sera
        acceptée. Les adresses de type contact@, ccas.ville@, mairie@,
        direction@, franceservices@, etc. ne sont donc pas acceptées. Nous
        acceptons des adresses emails créées pour l'occasion, qu'importe le
        fournisseur de messagerie électronique choisi.
      </>
    ),
    displayIndividualEmailLabel: true,
  },
  responsable_traitement: null,
  delegue_protection_donnees: null,
  responsable_technique: null,
  aidant: {
    header: 'Aidant',
    description: (
      <>
        <b>Les aidants</b> professionnels vont réaliser des démarches
        administratives à la place des usagers de façon sécurisée avec
        AidantsConnect.
      </>
    ),
    multiple: true,
    displayIndividualEmailLabel: true,
    contactByEmailOnly: true,
  },
};

const additionalTermsOfUse = [
  {
    id: 'has_professional_contact_only',
    label: (
      <>
        Je confirme que la liste des aidants à habiliter contient exclusivement
        des aidants professionnels. Elle ne contient donc ni service civique, ni
        bénévole, ni apprenti, ni stagiaire.
      </>
    ),
  },
  {
    id: 'has_non_elected_contact_only',
    label: (
      <>
        Je confirme qu’aucun élu n’est impliqué dans l’habilitation Aidants
        Connect. Le responsable Aidants Connect ainsi que les aidants à
        habiliter ne sont pas des élus.
      </>
    ),
  },
];

const target_api = 'aidants_connect';

const AidantsConnect = () => (
  <Form
    target_api={target_api}
    contactEmail={DATA_PROVIDER_PARAMETERS[target_api]?.email}
    documentationUrl="https://aidantsconnect.beta.gouv.fr/"
  >
    <OrganisationSection />
    <StructureSection />
    <ÉquipeSection initialContacts={initialContacts} />
    <AidantsSection />
    <CguSection
      cguLink="https://aidantsconnect.beta.gouv.fr/cgu/"
      additionalTermsOfUse={additionalTermsOfUse}
    />
  </Form>
);

export default AidantsConnect;
